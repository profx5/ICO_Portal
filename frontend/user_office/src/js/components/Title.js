import React from 'react'

const Title = ({
    text,
    type,
    center
}) => {
    
    const styles = (center) => {
        return {
            textAlign: center && 'center' || 'left'
        }
    }   

    switch(type) {
        case 'h1': {
            return  <h1 className="Header_head" style={styles(center)}>{text}</h1>
        }
        case 'h2': {
            return  <h2 className="Header_head" style={styles(center)}>{text}</h2>
        }
        case 'h3': {
            return  <h3 className="Header_head" style={styles(center)}>{text}</h3>
        }
        case 'h4': {
            return  <h4 className="Header_head" style={styles(center)}>{text}</h4>
        }
        default: {
            return  <span>{text}</span>
        }
    }
}

Title.defaultProps = {
    center: false
}

export default Title
