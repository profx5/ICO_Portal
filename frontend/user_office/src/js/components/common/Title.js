import React from 'react'
import styled from 'styled-components';
import {media} from './../../utils/media';

const Title = ({children, className}) => {
    return (
        <StyledTitle className={className}>{children}</StyledTitle>
    )
}


export default Title;

const StyledTitle = styled.h2`
    font-size: 38px;
    line-height: 1;
    font-weight: 400;
    color: #233539;
    letter-spacing: 0.8px;
    margin-top: 60px;
    flex-basis: 100%;
    &.Verification_head {
        margin-top: 0;
    }
    ${media.xs} {
        font-size: 20px;
        letter-spacing: 0.4px;
        margin-top: 10px;
    }
    ${media.sm} {
        font-size: 29px;
    }
`;
