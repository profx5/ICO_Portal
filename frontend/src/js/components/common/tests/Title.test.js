import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Title from 'js/components/common/Title';


configure({ adapter: new Adapter() });

it('should render stateless Title', () => {
    const component = shallow(<Title/>);
    expect(component).toMatchSnapshot();
})
