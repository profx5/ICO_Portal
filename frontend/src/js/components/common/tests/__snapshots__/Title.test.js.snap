// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless Title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "className": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    font-size: 38px;
    line-height: 1;
    font-weight: 400;
    color: #233539;
    letter-spacing: 0.8px;
    margin-top: 60px;
    flex-basis: 100%;
    white-space: nowrap;
    &.Verification_head {
        margin-top: 0;
    }
    ",
          "@media (max-width: 767px)",
          " {
        font-size: 20px;
        letter-spacing: 0.4px;
        margin-top: 10px;
    }
    ",
          "@media (min-width: 768px) and (max-width: 1024px)",
          " {
        font-size: 29px;
    }
",
        ],
      },
      "displayName": "styled.h2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h2",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "className": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    font-size: 38px;
    line-height: 1;
    font-weight: 400;
    color: #233539;
    letter-spacing: 0.8px;
    margin-top: 60px;
    flex-basis: 100%;
    white-space: nowrap;
    &.Verification_head {
        margin-top: 0;
    }
    ",
            "@media (max-width: 767px)",
            " {
        font-size: 20px;
        letter-spacing: 0.4px;
        margin-top: 10px;
    }
    ",
            "@media (min-width: 768px) and (max-width: 1024px)",
            " {
        font-size: 29px;
    }
",
          ],
        },
        "displayName": "styled.h2",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h2",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
